webapp/channels/src/components/user_settings/plugin/plugin_setting.tsx

```diff
@@ -87,6 +88,18 @@ const PluginSetting = ({
                     informChange={onSettingChanged}
                     pluginId={pluginId}
                 />);
+        } else if (setting.type === 'custom') {
+            const CustomComponent = setting.component;
+            const inputEl = (
+                <PluggableErrorBoundary
+                    key={setting.name}
+                    pluginId={pluginId}
+                    hideRefresh={true}
```

Comment:
Why hide the refresh button for this pluggable? I know that button is a bit ugly, but we're otherwise not giving users a way to try to get out of that error case

Reply:
@hmhealey Clicking refresh doesn't do anything. I tested a bit and thought that probably it wasn't designed for this use case. Also, since we are dealing with settings and configuration, I thought refreshing may not always be the best idea (losing data and such). To make it work (now it's a simple href to `#`) we'd probably have to use something like `location.reload();`.

Reply:
That name is actually a bit misleading. It just clears the error state instead of fully refreshing the page, so it'll try to render the component again. If the component errors out while rendering again, it'll just go back to the showing the error message like it was before.

You're not losing any data that's stored outside of the component at that point (in the modal, in redux, or anywhere outside of that), but any state stored inside of the pluggable would've been lost as soon as the error occurred and the error boundary unmounted the component

Reply:
Ah, interesting. Okay, that explains why a component throwing would be stuck in the same state. Not sure how helpful the refresh is in these cases, but I see your point. Let me update.